File: python_examples/src/service_server.py
#!/usr/bin/env python

import rospy
from python_examples.srv import SumaDosEnteros, SumaDosEnterosResponse

def suma_callback(req):
    rospy.loginfo("Primer numero: %d, Segundo numero: %d", req.a, req.b)
    return SumaDosEnterosResponse(req.a + req.b)

def service_server():
    rospy.init_node('service_server_suma')
    server_srv = rospy.Service('/SumaDosNumeros', SumaDosEnteros, suma_callback)  # Creación del servidor
    rospy.loginfo("Listo para mandar el resultado de la suma.")
    rospy.spin()

if __name__ == '__main__':
    service_server()

File: c_plus_plus_examples/src/service_server.cpp
#include <ros/ros.h>
#include <c_plus_plus_examples/SumaDosEnteros.h>

bool sumaCallback(c_plus_plus_examples::SumaDosEnteros::Request &req, c_plus_plus_examples::SumaDosEnteros::Response &res){
    ROS_INFO_STREAM("Primer numero: " << req.a << " Segundo numero: " << req.b);
    res.suma = req.a + req.b;

    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "service_server_suma");
    ros::NodeHandle nh;

    ros::ServiceServer server_srv = nh.advertiseService("/SumaDosNumeros", sumaCallback); //Creacion del server

    ROS_INFO("Listo para mandar el resultado de la suma.");
    ros::spin();

    return 0;
}